pp_hot.c

```diff
@@ -5357,8 +5357,7 @@ PP(pp_subst)
     if (dstr) {
         /* replacement needing upgrading? */
         if (DO_UTF8(TARG) && !doutf8) {
-             nsv = sv_newmortal();
-             SvSetSV(nsv, dstr);
+             nsv = sv_mortalcopy_flags(dstr, SV_GMAGIC|SV_DO_COW_SVSETSV);
```

Comment:
Re-think this chunk, macro `SvSetSV` macro has an ultra fast shortcut/bypass in it. That shortcut will be lost if this chinkgoes to production. `sv_mortalcopy_flags` is a linkable extern C func. The macro's shortcut path doesn't have any fn calls in it.

Reply:
That shortcut/bypass:
```
#define SvSetSV_and(dst,src,finally) \
        STMT_START {                                    \
            SV * src_ = src;                            \
            SV * dst_ = dst;                            \
            if (LIKELY((dst_) != (src_))) {             \
                sv_setsv(dst_, src_);                   \
                finally;                                \
            }                                           \
        } STMT_END
...
#define SvSetSV(dst,src) \
                SvSetSV_and(dst,src,/*nothing*/;)
```
`dst_` is never equal to `src_` here (unless `dst_` points to freed SV, which would be a serious bug and shouldn't be optimized for).

Reply:
> That shortcut/bypass:
....
> `dst_` is never equal to `src_` here (unless `dst_` points to freed SV, which would be a serious bug and shouldn't be optimized for).

@tonycoz you are correct, That macro's optimization branch is impossible to execute at this particular line.